code/datums/gamemode/dynamic/dynamic_rulesets_roundstart.dm

```diff
@@ -348,23 +348,25 @@
 	return (head_check >= required_heads)
 
 /datum/dynamic_ruleset/roundstart/revs/execute()
-	var/datum/faction/revolution/R = find_active_faction_by_type(/datum/faction/revolution)
-	if (!R)
-		R = ticker.mode.CreateFaction(/datum/faction/revolution, null, 1)
-
-	var/max_canditates = 4
-	for(var/i = 1 to max_canditates)
-		if(candidates.len <= 0)
-			break
-		var/mob/M = pick(candidates)
-		assigned += M
-		candidates -= M
-		var/datum/role/revolutionary/leader/lenin = new
-		lenin.AssignToRole(M.mind, 1, 1)
-		R.HandleRecruitedRole(lenin)
-		lenin.Greet(GREET_ROUNDSTART)
-	update_faction_icons()
-	return 1
+	spawn (5 MINUTES)
+		var/datum/faction/revolution/R = find_active_faction_by_type(/datum/faction/revolution)
+		if (!R)
+			R = ticker.mode.CreateFaction(/datum/faction/revolution, null, 1)
+
+		var/max_canditates = 4
+		for(var/i = 1 to max_canditates)
+			if(candidates.len <= 0)
+				break
+			var/mob/M = pick(candidates)
+			assigned += M
+			candidates -= M
+			var/datum/role/revolutionary/leader/lenin = new
+			lenin.AssignToRole(M.mind, 1, 1)
+			R.HandleRecruitedRole(lenin)
+			lenin.Greet(GREET_ROUNDSTART)
+		update_faction_icons()
+		R.OnPostSetup()
+		return 1
```

Comment:
Tab this back once, as execute is expected to return 1, else the ruleset is dumped.